test/e2e/flask/solana/common-solana.ts

```diff
@@ -41,22 +226,47 @@ export async function withSolanaAccountSnap(
   {
     title,
     solanaSupportEnabled,
-  }: { title?: string; solanaSupportEnabled?: boolean },
+    showNativeTokenAsMainBalance,
+    mockCalls,
+    mockSendTransaction,
+  }: {
+    title?: string;
+    solanaSupportEnabled?: boolean;
+    showNativeTokenAsMainBalance?: boolean;
+    mockCalls?: boolean;
+    mockSendTransaction?: boolean;
+  },
   test: (driver: Driver, mockServer: Mockttp) => Promise<void>,
 ) {
   console.log('Starting withSolanaAccountSnap');
+  let fixtures = new FixtureBuilder().withPreferencesControllerAndFeatureFlag({
```

Comment:
this method: 

```
  withPreferencesControllerAndFeatureFlag(flags) {
    merge(this.fixture.data.PreferencesController, flags);
    return this;
  }
```
is exactly the same as:

```
  withPreferencesController(data) {
    merge(this.fixture.data.PreferencesController, data);
    return this;
  }
```

Could we consider get rid of it, to avoid complexity and just use the withPreferenceController one?

Reply:
I removed it and changed where it was used. I did not add any of those though,  will keep an eye to see if any test fails (it shouldn't)